src/workerd/jsg/jsg.h

```diff
@@ -1770,6 +1770,44 @@ template <typename T>
 constexpr bool isV8MaybeLocal() { return isV8MaybeLocal((T*)nullptr); }
 
 class AsyncContextFrame;
+class JsValue;
+class JsObject;
+class JsBoolean;
+class JsArray;
+template <typename T> class JsRef;
+
+#define JS_V8_SYMBOLS(V) \
+  V(AsyncIterator) \
+  V(HasInstance) \
+  V(IsConcatSpreadable) \
+  V(Iterator) \
+  V(Match) \
+  V(Replace) \
+  V(Search) \
+  V(Split) \
+  V(ToPrimitive) \
+  V(ToStringTag) \
+  V(Unscopables)
+
+#define JS_TYPE_CLASSES(V) \
```

Comment:
These are translated into additional `Js{Type}` types, e.g. `JsString`, `JsDate`, etc... all of which are trivially assignable to/from `JsValue`. They exist only to facilitate the typescript/rtti type generation.